### PR TITLE
Node W1 HW Eyad

### PR DESCRIPTION
**Pull request by ekhamis**
Eyad HTTP

### CODE REVIEW

File: week1/index.js [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
+            counter = 10;
+            console.log('Counter value after RESET:', counter, '\n')
+            response.write(`<h2>Counter value was reseted to: ` + counter + `</h2>`, `utf8`);
+            break;
+        default:
+            response.statusCode = 404;
+            response.write(`<h2 style="color: red">Your request is invalid, please type either add, remove, state or reset to continue...<h2>`);
+            console.error('Error \n')
+    }
+    response.end();
```
Comment (rts-rob) [inline week1/index.js]: Well done - exactly the right place for something that only executes once but _always_ executes.

File: week1/index.js [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
+            response.write(`<h2>The counter was successfully decreased by 1. <br> Counter new value is: ` + counter + `</h2>`, `utf8`);
+            break;
+        case '/reset':
+            counter = 10;
+            console.log('Counter value after RESET:', counter, '\n')
+            response.write(`<h2>Counter value was reseted to: ` + counter + `</h2>`, `utf8`);
+            break;
+        default:
+            response.statusCode = 404;
+            response.write(`<h2 style="color: red">Your request is invalid, please type either add, remove, state or reset to continue...<h2>`);
```
Comment (rts-rob) [inline week1/index.js]: What's something else we could do here to make it easy for our users? They're in a browser, why type something when they could just click...

File: week1/index.js [hunk 3]
```diff
@@ -1,25 +1,42 @@
...
-server.listen(port, function(error) {
-  if (error) {
-    console.log(error);
-  } else {
-    console.log('api listening on port', port);
-  }
+    switch (request.url) {
+        case '/state':
+        case '/':
+            console.log('Original counter value:', counter, '\n')
```
Comment (rts-rob) [inline week1/index.js]: Try something:
1. Go to `/state`
2. Go to `/add`
3. Go back to `/state` (as a new request, not by hitting the back button

What gets logged to the console? Is that correct/true?

Comment (ekhamis) [inline week1/index.js]: good catch. Corrected and will be pushed soon :)

File: week1/index.js (hunk 2 above)
Comment (ekhamis) [inline week1/index.js]: coming up :)

File: week1/index.js (hunk 1 above)
Comment (ekhamis) [inline week1/index.js]: :)

File: week1/index.js [hunk 4]
```diff
@@ -1,25 +1,60 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
 var http = require('http');
```
Comment (daanaerts) [inline week1/index.js]: Hi Eyad,

- Great indentation, easy to read
- Question to think about: when to use `const`, `var`, or `let`?

File: week1/index.js [hunk 5]
```diff
@@ -1,25 +1,60 @@
...
+    response.setHeader('Content-Type', 'text/html');
+    console.log('The new received http request is: ', request.url)
 
-// Start the HTTP server, start listening for requests
-server.listen(port, function(error) {
-  if (error) {
-    console.log(error);
-  } else {
-    console.log('api listening on port', port);
-  }
```
Comment (daanaerts) [inline week1/index.js]: Interesting flow to handle request/response. I like it!

File: week1/index.js (hunk 4 above)
Comment (ekhamis) [inline week1/index.js]: Thanks noted. It was more of copy/paste blind spot, I have updated the code accordingly

**Pull request closed without merging**